### PR TITLE
[receiver/mongodbatlasreceiver] add provider resource attributes

### PR DESCRIPTION
**Pull request by JonathanWamsley**
**Description:** <Describe what has changed.>

This feature adds provider resource attributes `mongodb_atlas.provider.name` and `mongodb_atlas.region.name` to add additional context and filtering capabilities.

**Link to tracking Issue:** <Issue number if applicable>
#28833 

**Testing:** <Describe what testing was performed and which tests were added.>
Test were automatically updated. Live testing was performed and validated on clusters.
**Documentation:** <Describe the documentation added.>
Docs were automatically updated.

### CODE REVIEW

**Comment by mx-psi** [comment]
cc @schmikei

File: receiver/mongodbatlasreceiver/receiver.go [hunk 1]
```diff
@@ -219,6 +236,8 @@ func (s *mongodbatlasreceiver) extractProcessDatabaseMetrics(
 		rb.SetMongodbAtlasProcessTypeName(process.TypeName)
 		rb.SetMongodbAtlasProcessID(process.ID)
 		rb.SetMongodbAtlasDbName(db.DatabaseName)
+		rb.SetMongodbAtlasRegionName(providerValues.RegionName)
```
Comment (schmikei) [inline receiver/mongodbatlasreceiver/receiver.go:239]: Not intending to scope creep should we probably add this attribute to some forms of logs that have access to the cluster ProviderSettings as well? 

https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/f509060a8d1ab5ca4b5827e0c60d1149e3059908/receiver/mongodbatlasreceiver/access_logs.go#L307

https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/ae6d36bc4001f539433e6b03fcff365d9489bf4d/receiver/mongodbatlasreceiver/logs.go#L151

**Review by schmikei — CHANGES_REQUESTED** [review]
Looks good mostly, do have one question around whether we want to introduce these optional attributes for some form of log telemetry this receiver collects as well?

API endpoint doc for onlookers showing the new fields being included into the receiver: https://www.mongodb.com/docs/atlas/reference/api-resources-spec/v1/#tag/Clusters/operation/listLegacyClusters

File: receiver/mongodbatlasreceiver/receiver.go (hunk 1 above)
Comment (JonathanWamsley) [reply receiver/mongodbatlasreceiver/receiver.go:239]: Got it, added the provider and region name to the logs and updated the tests accordingly. Good suggestions.

Comment (schmikei) [reply receiver/mongodbatlasreceiver/receiver.go:239]: @JonathanWamsley seems like some test failures?

Comment (JonathanWamsley) [reply receiver/mongodbatlasreceiver/receiver.go:239]: ahh, my bad.  I updated the mock and golden file for the access logs integration test.

File: .chloggen/mongodbatlasreceiver-add-provider-resource-attributes.yaml [hunk 2]
```diff
@@ -0,0 +1,27 @@
+# Use this changelog template to create an entry for release notes.
+
+# One of 'breaking', 'deprecation', 'new_component', 'enhancement', 'bug_fix'
+change_type: enhancement
+
+# The name of the component, or a single word describing the area of concern, (e.g. filelogreceiver)
+component: receiver/mongodbatlasreceiver
+
+# A brief description of the change.  Surround your text with quotes ("") if it needs to start with a backtick (`).
+note: emit resource attributes "`mongodb_atlas.region.name`" and "`mongodb_atlas.provider.name`" on metric scrape.
```
Comment (schmikei) [inline .chloggen/mongodbatlasreceiver-add-provider-resource-attributes.yaml:10]:
```suggestion
note: emit optional resource attributes "`mongodb_atlas.region.name`" and "`mongodb_atlas.provider.name`".
```

**Review by schmikei — APPROVED** [review]
Looks mostly good to me except we now have a slightly out of date changelog.

@djaglowski when you have a moment, mind giving it a second pass?

**Comment by JonathanWamsley** [comment]
@djaglowski If you got a moment to review this, that would be great!

**Review by mx-psi — APPROVED** [review]
LGTM. I will merge by Thursday EU morning if no concerns are raised

cc @djaglowski

**Pull request merged**